keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/jackson/ImageProviderDeserializer.kt

```diff
@@ -35,4 +35,4 @@ internal class ImageProviderDeserializer :
     }
 }
 
-class InvalidPayload(strategy: String) : UserException("Payload is not valid for $strategy")
+class InvalidPayload(strategy: String) : SystemException("Payload is not valid for $strategy")
```

Comment:
Sorry, I should have added a comment since I went from one extreme to the other. This is only thrown by the custom polymorphic deserialization code @robfletcher wrote, which is why I thought it'd be better represented as a system exception. Rob, what was your rationale to use `UserException` originally?

Reply:
I think this happens in the case where you've submitted a resource with an invalid `imageProvide`, in which case I think this a `UserException`

Reply:
Isn't this used on submission of a delivery config? e.g. if your resource kind doesn't match the spec?

Reply:
Yes. I was gonna say "but also when we load it from the db", but that's not a thing. Will revert.